### ISSUE TITLE
Multi-select causes first selection to be duplicated

### ISSUE THREAD

**Issue by GrandPyromania** [yatki/vscode-surround#108]
Hi!

When I select multiple lines and use surround with, it causes the first line to be duplicated across all of the selections. Here's what I mean:

![Code_rvmp4SPSXR](https://github.com/yatki/vscode-surround/assets/10426305/f7494270-91e0-47d8-86f2-c1fe3f05062c)

Is there a way to fix this so that each selection retains its original text?

Thanks!

**Comment by yatki** [comment]
Hey @GrandPyromania thanks for reporting this. I am looking into it. 👍🏻

**Comment by yatki** [comment]
Hey @GrandPyromania this issue is not an easy fix and caused by vscode internal methods. I opened a ticket for it. let's see what they will say. There are workarounds but I would prefer to fix it properly.

**Comment by GrandPyromania** [comment]
Thank you!

**Comment by fhurta** [comment]
This is bad, just found out this plugin and used the multiselection for tagging articles..  Haven't noticed all tagged texts got replaced with the same text until final review...  You could perhaps disable using it with multi-selection until it works again..

The idea is perfect - I was looking exactly for such a tool. But now I'm little sad, the time investment into "work smarter, not harder", didn't really pay off :)

**Comment by yatki** [comment]
@fhurta Yeah, sorry for the inconvenience but this is unfortunately a vscode limitation. I will try to find a workaround.
